src/NodeAnalyzer/ScopeAnalyzer.php

```diff
@@ -49,6 +49,10 @@ public function resolveScope(
             return $this->scopeFactory->createFromFile($filePath);
         }
 
+        if ($node->getAttribute(AttributeKey::EXPRESSION_DEPTH) >= 2) {
+            return $this->scopeFactory->createFromFile($filePath);
+        }
```

Comment:
@TomasVotruba this is to avoid too deep scope filing assign -> assign -> expr -> method call -> property fetch , eg:

```php
$$param = $$bar = self::decodeValue($result->getItem()->getTextContent());
```